File: src/main/java/com/DavidM1A2/afraidofthedark/common/spell/component/InvalidValueException.java
package com.DavidM1A2.afraidofthedark.common.spell.component;

/**
 * An exception that gets thrown when an editable spell component property gets an invalid value
 */
public class InvalidValueException extends RuntimeException
{
    /**
     * Constructor that takes in an error message to be displayed to the user
     *
     * @param message the detail message
     */
    public InvalidValueException(String message)
    {
        super(message);
    }
}

File: src/main/kotlin/com/davidm1a2/afraidofthedark/common/spell/component/InvalidValueException.kt
package com.davidm1a2.afraidofthedark.common.spell.component

import net.minecraft.util.text.ITextComponent

/**
 * An exception that gets thrown when an editable spell component property gets an invalid value
 *
 * @constructor that takes in an error message to be displayed to the user
 * @param reason the detail message
 */
class InvalidValueException(val reason: ITextComponent) : RuntimeException()
